### ISSUE TITLE
Overly strict "Python tag" handling for "abi3"-tagged wheels

### ISSUE THREAD

**Issue by exarkun** [DavHau/mach-nix#464]
https://peps.python.org/pep-0384/ defines the "Stable ABI" for CPython.  It claims that all subsequent CPython 3.x releases will support this ABI.

Then https://peps.python.org/pep-3149/ and https://peps.python.org/pep-0425/ expand on this to define the "abi3" "ABI Tag" as referring to the "Stable ABI" - ie, says that wheels with an "ABI Tag" of "abi3" are built against the "Stable ABI".

*Then* https://discuss.python.org/t/python-tags-specific-version-of-interpreter-or-minimum-version/7831/27 suggests that inference on any of this is silly, just compare a wheel's tags to the elements of `packaging.tags.sys_tags()` and if there's a match, you've got a wheel you can use.

mach-nix will only use a wheel with a "Python tag" that exactly matches the version of Python it is using though.  Compared to what pip will do, this excludes many wheels (for example, all `cryptography` wheels released in over a wheel).

Should mach-nix do something differently so it can consume more wheels?

**Comment by DavHau** [comment]
Oh thanks for that. This should definitely be fixed in mach-nix.

**Comment by DavHau** [comment]
>  Do you have any sense of whether it makes sense to try to use packaging.tags.sys_tags() in mach-nix to fix this?

I don't think that doing this is a good idea, as it would exclude wheels which are built against a later python version than the one of the current interpreter.

Though #471 should fix the problem. Could you check?

**Comment by exarkun** [comment]
#471 was reverted in 552d4caa73722b262204319526f9e77f9370f702 due to regression reported in #475 so this issue should be open, I think (and it seems like #475 could be closed).
